maas/resource_maas_dns_record_test.go

```diff
@@ -0,0 +1,201 @@
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testutils.PreCheck(t, nil) },
+		Providers:    testutils.TestAccProviders,
+		CheckDestroy: testAccMAASDNSRecordCheckDestroy(testIPAddress),
+		ErrorCheck:   func(err error) error { return err },
+		Steps: []resource.TestStep{
+			{
+				Config: getDNSRecordConfigBasic(recordName, testRecordType, testIPAddress, testDomain),
+				Check: resource.ComposeTestCheckFunc(
+					testAccMAASDNSRecordCheckExists("maas_dns_record."+resourceName, &dnsRecord),
+					resource.TestCheckResourceAttr("maas_dns_record."+resourceName, "name", recordName),
+					resource.TestCheckResourceAttr("maas_dns_record."+resourceName, "type", testRecordType),
+					resource.TestCheckResourceAttr("maas_dns_record."+resourceName, "data", testIPAddress),
+					resource.TestCheckResourceAttr("maas_dns_record."+resourceName, "domain", testDomain),
+				),
+			},
+		},
+	})
+}
+
+// Test that two DNS records with the same IP address can be created and destroyed.
+func TestAccResourceMAASDNSRecord_sameIPAddress(t *testing.T) {
+	var dnsRecord entity.DNSResource
+	recordName1 := acctest.RandomWithPrefix("tf-1-")
+	recordName2 := acctest.RandomWithPrefix("tf-2-")
+	resourceName1 := "test_aaaa_1"
+	resourceName2 := "test_aaaa_2"
+	testIPAddress := "8.8.8.9"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testutils.PreCheck(t, nil) },
+		Providers:    testutils.TestAccProviders,
+		CheckDestroy: testAccMAASDNSRecordCheckDestroy(testIPAddress),
+		ErrorCheck:   func(err error) error { return err },
+		Steps: []resource.TestStep{
+			{
+				Config: getDNSRecordConfigSameIPA_AAAA(resourceName1, resourceName2, recordName1, recordName2, testIPAddress),
+				Check: resource.ComposeTestCheckFunc(
+					testAccMAASDNSRecordCheckExists("maas_dns_record."+resourceName1, &dnsRecord),
+					testAccMAASDNSRecordCheckExists("maas_dns_record."+resourceName2, &dnsRecord),
+					resource.TestCheckResourceAttr("maas_dns_record."+resourceName1, "name", recordName1),
+					resource.TestCheckResourceAttr("maas_dns_record."+resourceName2, "name", recordName2),
+					resource.TestCheckResourceAttr("maas_dns_record."+resourceName1, "type", "A/AAAA"),
+					resource.TestCheckResourceAttr("maas_dns_record."+resourceName2, "type", "A/AAAA"),
+					resource.TestCheckResourceAttr("maas_dns_record."+resourceName1, "data", testIPAddress),
+					resource.TestCheckResourceAttr("maas_dns_record."+resourceName2, "data", testIPAddress),
+					resource.TestCheckResourceAttr("maas_dns_record."+resourceName1, "domain", "maas"),
+					resource.TestCheckResourceAttr("maas_dns_record."+resourceName2, "domain", "maas"),
+				),
+			},
+		},
+	})
+}
+
+func getDNSRecordConfigSameIPA_AAAA(resourceName1 string, resourceName2 string, recordName1 string, recordName2 string, ipAddress string) string {
+	return fmt.Sprintf(`
+	resource "maas_dns_record" %q {
+	  name   = %q
+	  type   = "A/AAAA"
+	  data   = %q
+	  domain = "maas"
```

Comment:
let's create a random domain as part of the tests since there is no guarantee that MAAS under test is using it. The change is trivial and should make the CI happy :slightly_smiling_face: Same thing applies to all places where `maas` domain is considered as something that preexists.